lib/internal/util/inspect.js

```diff
@@ -1319,7 +1319,9 @@ function getStackString(ctx, error) {
     if (typeof error.stack === 'string') {
       return error.stack;
     }
-    return formatValue(ctx, error.stack);
+    if (error.stack !== error) {
+      return formatValue(ctx, error.stack);
+    }
```

Comment:
```suggestion
    ctx.seen.push(error);
    const result = formatValue(ctx, error.stack);
    ctx.seen.pop();
    return `${ErrorPrototypeToString(error)}\n    ${result.replaceAll('\n', '\n    ')}`;
```

This is going to catch all circular issues and it provides additional information to the user. It does require tests to be adjusted accordingly but only a few.

Reply:
This actually looks great! Thanks for the suggestion!